app/src/main/java/chat/rocket/android/chatroom/ui/bottomsheet/WebUrlBottomSheet.kt

```diff
@@ -0,0 +1,116 @@
+package chat.rocket.android.chatroom.ui.bottomsheet
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.FrameLayout
+import chat.rocket.android.R
+import chat.rocket.android.util.extensions.ui
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.android.synthetic.main.app_bar.*
+import kotlinx.android.synthetic.main.webview_bottomsheet.*
+
+private const val BUNDLE_WEB_PAGE_URL = "web_page_url"
+private const val BUNDLE_CHATROOM_ID = "chatroom_id"
+private const val BUNDLE_HEIGHT_RATIO = "height_ratio"
+
+class WebUrlBottomSheet : BottomSheetDialogFragment() {
+    private lateinit var webPageUrl: String
+    private lateinit var chatRoomId: String
+    private lateinit var heightRatio: String
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        arguments?.run {
+            webPageUrl = getString(BUNDLE_WEB_PAGE_URL, "")
+            chatRoomId = getString(BUNDLE_CHATROOM_ID, "")
+            heightRatio = getString(BUNDLE_HEIGHT_RATIO, "")
+        } ?: requireNotNull(arguments) { "no arguments supplied when the fragment was instantiated" }
+        return inflater.inflate(R.layout.webview_bottomsheet, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        setupToolbar()
+        setupWebView()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val bottomsheet: FrameLayout = dialog.findViewById(com.google.android.material.R.id.design_bottom_sheet)
+        val behaviour = BottomSheetBehavior.from(bottomsheet)
+        behaviour.setBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback() {
+            override fun onSlide(p0: View, p1: Float) {
+            }
+
+            override fun onStateChanged(p0: View, newState: Int) {
+                if (newState == BottomSheetBehavior.STATE_HIDDEN) {
+                    dialog.cancel()
+                }
+                if(heightRatio == "tall" && newState == BottomSheetBehavior.STATE_DRAGGING){
+                    behaviour.state = BottomSheetBehavior.STATE_COLLAPSED
+                }
+                if(heightRatio == "compact" && newState == BottomSheetBehavior.STATE_EXPANDED){
+                    behaviour.isHideable = false
+                }
+            }
+
+        })
+        when(heightRatio){
```

Comment:
What about the `full` here? Is it not needed?

Reply:
@filipedelimabrito `full` is not needed here because only `tall` and `compact` configurations are using bottomsheet and `full` configuration requires a new Activity i.e webviewActivity. `full` doesn't need a bottomsheet.